### DISCUSSION TITLE
[Connector Request] Wallet Standard connector

### DISCUSSION THREAD

**Discussion by bulgakovk** [wagmi-dev/references#273] [Connector request]
Hey there! Opening a discussion to propose adding a new Wagmi Connector according to the [guideline](https://github.com/wagmi-dev/wagmi/blob/main/.github/CONTRIBUTING.md).

**What novel use-case does the Connector provide?**

Wallet Standard (Standard) connector provides a connector for Browser Extension Wallets registering their providers in the [Standard](https://github.com/wallet-standard/wallet-standard) compatible way. The Connector significantly improves the discoverability aspect of wallets implementing [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) API:

1. Since Standard wallets do not compete for the `window.ethereum` name space, multiple wallets can co-exist, which significantly improves UX. 
2. Standard wallets implement  an  event based API for [registering a wallet](https://github.com/wallet-standard/wallet-standard/blob/master/packages/core/wallet/src/register.ts#L25) for wallet developers and [detecting registered wallets](https://github.com/wallet-standard/wallet-standard/blob/8ac98a8ac6001a34906d01eac2ae0825f7208b32/packages/example/wallets/src/window.ts#L20-L25) for app developers. That eliminates the race condition issue when a Dapp tries to access `window.ethereum` before the property was defined.

**Are the Connector's integrations production-ready and generally available?**

1. Wallet Standard (Rainbowkit + Wagmi) playground is available here: https://github.com/ExodusMovement/wallet-standard-rainbowkit
2. `@wallet-standard/wallet` (_a core package to be consumed by a wallet_) – has a stable v1 release and publicly available at https://github.com/wallet-standard/wallet-standard/tree/master/packages/core/wallet
3. `@wallet-standard/app` (a core package to be consumed by an app) – has a stable v1 release and publicly available at https://github.com/wallet-standard/wallet-standard/tree/master/packages/core/app
4. The [react package](https://github.com/wallet-standard/wallet-standard/tree/master/packages/react/core) is also publicly available.

I would be happy to contribute to the repository and maintain the connector in the future. Please let me know what do you think. Thanks!

**Comment by tmm** [comment]
Thanks for opening this up! Working on migrating wagmi from ethers to [viem](https://viem.sh) at the moment and will circle back soon.

**Comment by bulgakovk** [comment]
Hey @tmm ! Just wanted to let you know that I've migrated the PoC repository to rainbowkit/wagmi v1 to keep it up-to-date :)

**Reply by glitch-txs** [reply]
What's different there?

**Comment by tmm** [comment]
@bulgakovk How is this different than [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963)?

**Comment by bulgakovk** [comment]
@tmm Hey, thanks for getting back to this proposal! So, EIP-6963 mainly focuses on fixing the wallet collision issue, whereas the Wallet Standard goes beyond that. Here are a couple of key points to consider:

1. Wallet Standard is a chain agnostic protocol by the design. It's pretty awesome for multi-chain wallets because they can avoid repetitive code patterns and enjoy more flexibility.

2. The Wallet Standard is not only compatible with EIP-1993 (check out the [example here](https://github.com/wallet-standard/wallet-standard/blob/master/packages/example/eip1193/src/initialize.ts#L6)), but it also allows wallets to define their own [APIs and future enhancements](https://github.com/wallet-standard/wallet-standard/blob/master/packages/core/base/src/wallet.ts#L101) along with [Wallet Accounts API](https://github.com/wallet-standard/wallet-standard/blob/master/packages/core/base/src/wallet.ts#L130).

**Reply by glitch-txs** [reply]
Hi, how would you handle the integration with the provider with this solution? For example EVM providers will follow 1193, but what about the other chains?

**Reply by bulgakovk** [reply]
Hey @glitch-txs, that's a good question. This is an example of how it can be done for Solana: https://github.com/brave/wallet-standard-brave/blob/master/src/wallet.ts#L62-L99. Generally speaking, Wallet Standard is compatible with any injected interface by the design.

**Reply by glitch-txs** [reply]
So does Brave support this for EVM? Do you know more wallets that do?

**Reply by bulgakovk** [reply]
Phantom, Brave and Exodus currently support Wallet Standard for Solana. However, the main challenge lies in the connector libraries, as most apps are not willing to put in the effort to integrate the standard directly. So that's why we're here :)

**Reply by glitch-txs** [reply]
Sounds interesting! I'm working on a connectors library as well, added EIP-6963 support already, I'll take a look, thanks!

**Comment by bulgakovk** [comment]
Hey @tmm! Just wanted to follow up my response above. Is there something else I can help you with to make it clear what makes Wallet Standard different?

**Comment by bulgakovk** [comment]
Hey @tmm ! Any luck with an update on this request?

**Reply by tmm** [reply]
Hi! Appreciate the follow up. We are hesitant to include this as a core connector because it's unclear what the benefits are, especially with [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) on the horizon, and adding new connectors adds some overhead for the core team. In the meantime, you are welcome to publish it as a separate npm package and post about it in the discussion form and add to [awesome-wagmi](https://github.com/wagmi-dev/awesome-wagmi).
